syntax = "proto3";


option objc_class_prefix = "FD";

package fundamental_data;

service FundamentalData {
    rpc ProcessReport (Report) returns (Empty) {}
    rpc GetLatestFinancialReportDate (StockRequest) returns (ReportMetaData) {}
    rpc IsUpToDate (StockRequest) returns (UpToDate) {}
}

message Empty {}

message ReportRequest {
    string stock = 1;
    string reportType = 2;
    string content = 3;
}

message StockRequest{
    string stock = 1;
}
message ReportMetaData {
    string reportDate = 1;
    string insertDate = 2;
}

message UpToDate{
    bool upToDate = 1;
    // add other information later on
}